### PR TITLE
[Feature] Inputd Mods and Fixes

### PR DESCRIPTION
**Pull request by Nevrdid**
- Add a system tool to change inputd version.
- Also avoid Terminal launch with a patched inputd

The Terminal is broken with patched inputd ( lags of hell) so i made this to easily switch between versions.

Credit to Tomato which made the patchs.
The main purpose of patch is to fix the same issue as d-pad some ones get with analogic sticks.
It skip a threshold check the original inpud made.
The 1ms patch also reduce the pooling rate. This reduce latency which is a must have for rythm games and probably speedrunners.

Update:
We made our own binaries instead of two old ones.

- new /var/trimui_inputd/swap_ab with it systool menu
- new /var/trimui_inputd/sticks_disabled which can be used in script to avoid sdl broken app as the Terminal
- Add polling rate choice
- Apply sticks_disabled when needed.

### CODE REVIEW

File: Apps/SystemTools/Menu/INPUTD (value)/.switcher.sh [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ln -s "$bin_dir$new_inputd" "$bin_dir/trimui_inputd"
+
+# Menu modification to reflect the change immediately
+
+# update crossmix.json configuration file
+json_file="/mnt/SDCARD/System/etc/crossmix.json"
+if [ ! -f "$json_file" ]; then
+	echo "{}" >"$json_file"
+fi
+jq --arg script_name "$script_name" '. += {"Inputd": $script_name}' "$json_file" >"/tmp/json_file.tmp" && mv "/tmp/json_file.tmp" "$json_file"
```
Comment (Kenny417) [inline Apps/SystemTools/Menu/INPUTD (value)/.switcher.sh:42]: I had to make the key in `crossmix.json` all caps for the menu item to be correct after rebooting the device. Like so: 

```suggestion
jq --arg script_name "$script_name" '. += {"INPUTD": $script_name}' "$json_file" >"/tmp/json_file.tmp" && mv "/tmp/json_file.tmp" "$json_file"
```

File: Apps/SystemTools/Menu/INPUTD (value)/Patched_14ms.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
+cd "$(dirname "$0")"
+
+source .switcher.sh
```
Comment (Kenny417) [inline Apps/SystemTools/Menu/INPUTD (value)/Patched_14ms.sh:4]: This line in all three scripts was not working on my device. I had to modify it like so: 

```suggestion
source "$(pwd)/.switcher.sh"
```

Something in my environment was causing confusion with the relative path. Adding the full working directory fixed it on my device.

**Review by Kenny417 — COMMENTED** [review]
I know I'm not a dev on this project but some folks on Reddit have shown interest in this patch. I don't know if there are any plans to merge this into CrossMix-OS but I took some time to test it out on my device. 

I found two problems in my testing but once those were resolved, this seems to work great. I hope this can get polished up a bit and merged into CrossMix some day.

File: Apps/SystemTools/Menu/INPUTD (value)/.switcher.sh (hunk 1 above)
Comment (Nevrdid) [reply Apps/SystemTools/Menu/INPUTD (value)/.switcher.sh:42]: I myself use Inputd in folder name on my sdcard as i never noticed it ^^

File: Apps/SystemTools/Menu/INPUTD (value)/Patched_14ms.sh (hunk 2 above)
Comment (Nevrdid) [reply Apps/SystemTools/Menu/INPUTD (value)/Patched_14ms.sh:4]: I honestly don't see any reason for this to fix an issue but if it really fix it for you i guess i can use the absolute path. Thanks.

**Comment by cizia64** [comment]
Hi, that's very nice, I suggest some modifications about the place of the script inside SYSTEM TOOLS : 
I think that it would be great to have a new "CONTROLS" sections which will contain these new scripts and the old "LEDS" scripts.

`Menu -> CONTROLS -> D-PAD FIX -> origin.sh`

May be later we will have an option to invert A and B in this section too.

What do you think about this ?

Also a little question: why a symbolic link instead of the binary itself ?

**Comment by Nevrdid** [comment]
> Hi, that's very nice, I suggest some modifications about the place of the script inside SYSTEM TOOLS : I think that it would be great to have a new "CONTROLS" sections which will contain these new scripts and the old "LEDS" scripts.
> 
> `Menu -> CONTROLS -> D-PAD FIX -> origin.sh`
> 
> May be later we will have an option to invert A and B in this section too.
> 
> What do you think about this ?

:+1:  

> Also a little question: why a symbolic link instead of the binary itself ?

Just because it allow to know which exact version the inputd in use refer to and so to know just looking files if system is set to original or patched easily as i did to check before launching terminal :
`custom_intputd=$(ls -la /usr/trimui/bin | grep "trimui_inputd -> /usr/trimui/bin/trimui_inputd_patched")`

**Comment by Nevrdid** [comment]
Update: 
No more use the old two patch but a new binary instead.
- new /var/trimui_inputd/swap_ab with it systool menu
- new /var/trimui_inputd/sticks_disabled which can be used in script to avoid sdl broken app as the Terminal
- Update system's inputd at started

Todo:
- Add polling rate choice
- apply sticks_disabled when needed.

**Review by cizia64 — APPROVED** [review]
Enormous work on these control features, thank you !

**Pull request merged**